README.md

```diff
@@ -1,29 +1,78 @@
+new_newsletter  GET     /newsletters/new(.:format)      newsletters#new
+edit_newsletter GET     /newsletters/:id/edit(.:format) newsletters#edit
+newsletter      GET     /newsletters/:id(.:format)      newsletters#show
+                PATCH   /newsletters/:id(.:format)      newsletters#update
+                PUT     /newsletters/:id(.:format)      newsletters#update
+                DELETE  /newsletters/:id(.:format)      newsletters#destroy
+```
+
+Rails does a great job of integrating RESTful routes into their system, if you can understand routes in Rails you can understand REST in general. Going through the output above you should be able to see that all of the potential CRUD actions are all there, from querying all of the records to deleting a single item from the database, and all of the actions are wired up using RESTful routing nomenclature.
+
+Here is a diagram that shows how how the views, controller actions, routes, and HTTP verbs are all mapped together
+
+![REST Diagram](http://reif.io/lib/flatiron/rest_diagram.png)
+
+In analyzing the diagram, you can see the flow of data as follows:
+
+1. The HTTP request contains a HTTP verb and hits a specific URL path
+
+2. The router in the application processes the request and 'routes' the request data to the proper controller action
+
+3. The controller action either performs a task, such as creating, updating, or deleting a record in the database, or it runs a database query and renders a view to the client
+
+## Definition of HTTP Verbs
+
+So what do those 'GET', 'POST', et al items represent? Those are HTTP verbs that each give the HTTP request unique behavior, below is an explanation on each verb:
+
+* **GET** - The GET method means retrieve whatever information (in the form of an entity) is identified by the Request-URI. This means if you go to /posts you will get all of the posts that the application wants your application to have.
+
+* **POST** - The POST method is used to request that the origin server accept the entity enclosed in the request as a new subordinate of the resource identified by the Request-URI in the Request-Line. This means that the server needs to accept the data sent over with the request.
+
+* **PATCH/PUT** - The PUT method requests that the enclosed entity be stored under the supplied Request-URI. If the Request-URI refers to an already existing resource, the enclosed entity SHOULD be considered as a modified version of the one residing on the origin server. This means that the server needs to accept data and that it is understood that the data is going to modify a pre-existing element in the database.
+
+* **DELETE** - The DELETE method requests that the origin server delete the resource identified by the Request-URI. This means… it deletes the record, it's nice and explicit.
+
+
+## Summary
+
+Below are a few keys to remember when thinking about REST:
+
+* REST is an architectural design pattern, not a framework or code in itself.
```

Comment:
What does that mean in the real world? Other frameworks, implement the REST pattern. Rails is not unique in this manner